### PR TITLE
move common params check to class IndexCommonParam

### PR DESCRIPTION
**Pull request by LHT129**
- common params(dim,metric_type,data_type...) should be checked for all index
- add param value's datatype check for hnsw and common params

### CODE REVIEW

File: src/index/index_common_param.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    // Check MetricType
+    CHECK_ARGUMENT(params.contains(PARAMETER_METRIC_TYPE),
+                   fmt::format("parameters must contains {}", PARAMETER_METRIC_TYPE));
+    const auto metric_obj = params[PARAMETER_METRIC_TYPE];
+    CHECK_ARGUMENT(metric_obj.is_string(),
+                   fmt::format("parameters[{}] must string type", PARAMETER_METRIC_TYPE));
+    std::string metric = metric_obj;
+    if (metric == METRIC_L2) {
```
Comment (inabao) [inline src/index/index_common_param.cpp:58]: Perhaps we don't need to require strict string equality here; we can ignore case.

Comment (LHT129) [reply src/index/index_common_param.cpp:58]: we can discuss the strict equality on another pull request.

**Review by inabao — APPROVED** [review]
LGTM

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**